### PR TITLE
Better Error Messages, Cleanup of trace options, Misc Improvements

### PR DESCRIPTION
**Pull request by justin808**
- Errors raised are of type ReactOnRailsError.
- Unused `server_render_method` was removed from the configuration.
- Removed ReactOnRails::Utils.server_bundle_file_name and
  ReactOnRails::Utils.bundle_file_name.

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1059)

### CODE REVIEW

**Comment by mapreal19** [comment]
:lgtm: 
-(review needed) 
+(approved for merge) 
+@mapreal19 

---

Reviewed 23 of 24 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks broke.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/shakacode/react_on_rails/1059#-:-LADde9g7GY_S5P0WsHB:b-qloubf)*

File: lib/react_on_rails/configuration.rb [hunk 1]
```diff
@@ -156,7 +161,6 @@ def initialize(node_modules_location: nil, server_bundle_js_file: nil, prerender
       self.webpack_generated_files = webpack_generated_files
       self.rendering_extension = rendering_extension
 
-      self.server_render_method = server_render_method
```
Comment (justin808) [inline lib/react_on_rails/configuration.rb]: Add a message, if the value is provided that it's been removed and to contact ShakaCode.com for React on Rails Pro.

**Comment by JPWallhorn** [comment]
Reviewed 14 of 24 files at r1, 59 of 60 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks broke.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/shakacode/react_on_rails/1059)*

**Comment by mapreal19** [comment]
Reviewed 25 of 61 files at r2, 143 of 145 files at r3, 8 of 63 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[lib/react_on_rails/react_on_rails_helper.rb, line 233 at r3](https://beta.reviewable.io/reviews/shakacode/react_on_rails/1059#-LAbGMW60NYPr_C7bHnb:-LAbGMW60NYPr_C7bHnc:b-o74r6y) ([raw file](https://github.com/shakacode/react_on_rails/blob/03a8c9e6b1be5232d3658ff3120c1ec918c032b9/lib/react_on_rails/react_on_rails_helper.rb#L233)):*
> ```Ruby
>       result = ReactOnRails::ServerRenderingPool.server_render_js_with_console_logging(wrapper_js)
> 
>       # IMPORTANT: To ensure that Rails doesn't auto-escape HTML tags, use Exception in rendering!the 'raw' method.
> ```

I think this comment should go to the docs. Like it's written now I don't understand it

---

*[spec/dummy/Gemfile, line 11 at r3](https://beta.reviewable.io/reviews/shakacode/react_on_rails/1059#-LAbJdigCkxZsWmsY_QU:-LAbJdigCkxZsWmsY_QV:b-fbxans) ([raw file](https://github.com/shakacode/react_on_rails/blob/03a8c9e6b1be5232d3658ff3120c1ec918c032b9/spec/dummy/Gemfile#L11)):*
> ```Ruby
> gem "webpacker"
> 
> # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
> ```

this comment should be before `gem "rails"` or just delete it since is not really helpful

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/shakacode/react_on_rails/1059)*

**Comment by mapreal19** [comment]
@justin808 let me know for the final review

**Comment by mapreal19** [comment]
Review status: 155 of 157 files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---

*[spec/dummy_no_webpacker/client/package.json, line 78 at r5](https://beta.reviewable.io/reviews/shakacode/react_on_rails/1059#-LAbbPwM9B-yPDAXa_Kv:-LAbbPwM9B-yPDAXa_Kw:beee1ph) ([raw file](https://github.com/shakacode/react_on_rails/blob/a781bc029944af0d931ac68ed7ae5aedf935d80f/spec/dummy_no_webpacker/client/package.json#L78)):*
> ```JSON
>     "build:server": "webpack --config webpack.server.rails.build.config.js",
>     "hot-assets": "babel-node server-rails-hot.js",
>     "install-react-on-rails": "rm -rf node_modules/react-on-rails && npm i 'file:../../..'"
> ```

CI fails bc can't run this one that you removed here

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/shakacode/react_on_rails/1059)*

**Comment by justin808** [comment]
Review status: 152 of 157 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[lib/react_on_rails/react_on_rails_helper.rb, line 233 at r3](https://beta.reviewable.io/reviews/shakacode/react_on_rails/1059#-LAbGMW60NYPr_C7bHnb:-LAbf1lr-iV-eBh91rgn:b-8h66te) ([raw file](https://github.com/shakacode/react_on_rails/blob/03a8c9e6b1be5232d3658ff3120c1ec918c032b9/lib/react_on_rails/react_on_rails_helper.rb#L233)):*
<details><summary><i>Previously, mapreal19 (Mario Pérez) wrote…</i></summary><blockquote>

I think this comment should go to the docs. Like it's written now I don't understand it
</blockquote></details>

me too...

must be something old

---

*[spec/dummy/Gemfile, line 11 at r3](https://beta.reviewable.io/reviews/shakacode/react_on_rails/1059#-LAbJdigCkxZsWmsY_QU:-LAbf9y06thLv7qWtS6L:bb7f2mf) ([raw file](https://github.com/shakacode/react_on_rails/blob/03a8c9e6b1be5232d3658ff3120c1ec918c032b9/spec/dummy/Gemfile#L11)):*
<details><summary><i>Previously, mapreal19 (Mario Pérez) wrote…</i></summary><blockquote>

this comment should be before `gem "rails"` or just delete it since is not really helpful
</blockquote></details>

done

---

*[spec/dummy_no_webpacker/client/package.json, line 78 at r5](https://beta.reviewable.io/reviews/shakacode/react_on_rails/1059#-LAbbPwM9B-yPDAXa_Kv:-LAbeEXC8Kpqros5SQBL:bcp1t5u) ([raw file](https://github.com/shakacode/react_on_rails/blob/a781bc029944af0d931ac68ed7ae5aedf935d80f/spec/dummy_no_webpacker/client/package.json#L78)):*
> ```
>   "install-react-on-rails": "rm -rf node_modules/react-on-rails && npm i 'file:../../..'"
> ```
and no postinstall!

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/shakacode/react_on_rails/1059)*

**Comment by mapreal19** [comment]
Reviewed 26 of 61 files at r2, 7 of 63 files at r4, 1 of 2 files at r5, 3 of 4 files at r6, 3 of 3 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/shakacode/react_on_rails/1059)*

**Comment by mapreal19** [comment]
Reviewed 8 of 8 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/shakacode/react_on_rails/1059)*

**Pull request merged**